Support Mac binary releases

Timmmm opened the pull request:
Add support to the release.yml Github Actions workflow for building a Mac binary. Unfortunately I got weird Git errors from OPAM when using `--filter=tree:0` so I had to remove it. Otherwise this is fairly simple.

I also upgraded the OCaml version to match build.yml.

jordancarlin commented on .github/workflows/release.yml:
```diff
@@ -49,23 +54,29 @@ jobs:
+    - name: System dependencies (Mac)
+      if: startsWith(matrix.os, 'macos')
+      run: |
+        brew install --force --overwrite gpatch gmp z3 pkg-config opam git
```
```suggestion
        brew install --force --overwrite gpatch gmp z3 pkgconf opam git
```

See #810

arichardson commented:
This looks like a great idea. I think the best thing would be if we could get sail into homebrew to make it easy to install.

I created a formula that probably no longer works a few years ago but no longer feel like upstreaming it since I had some less than ideal experiences dealing with homebrew... In case anyone would like to try here is my formula: https://github.com/arichardson/homebrew-cheri/blob/main/Formula/sail.rb

Commit:
Support Mac binary releases

Add support to the release.yml Github Actions workflow for building a Mac binary. Unfortunately I got weird Git errors from OPAM when using `--filter=tree:0` so I had to remove it. Otherwise this is fairly simple.

I also upgraded the OCaml version to match build.yml.
